corehq/util/archive_old_forms.py

```diff
@@ -0,0 +1,89 @@
+from corehq.apps.es import cases as case_es
+from corehq.apps.es.forms import FormES
+from corehq.form_processor.models.forms import XFormInstance
+import datetime
+from math import ceil
+import os
+
+# Note: There are over 2.3 million cases that need to be archived on the target project. For this reason,
+# batching will be necessary. Furthermore, the batch sizes are kept small as each case can have 2-3+ forms
+# that will need to be archived.
+
+# Constants
+DOMAIN = 'test-65' # 'fmoh-echis-staging'
+DATE_CONSTRAINT = datetime.date(2022, 5, 1)
+BATCH_SIZE = 100
+ERROR_LOG_FILE_PATH = os.path.expanduser("~/zeng_script_error.log")
+SUCCESS_LOG_FILE_PATH = os.path.expanduser("~/zeng/script_success.log")
+
+# Get the total number of cases to process
+total_cases = (
+    case_es.CaseES()
+    .domain(DOMAIN)
+    .OR(
+        case_es.is_closed(True),
+        case_es.opened_range(lt=DATE_CONSTRAINT)
+    )
+    .count()
+)
+batch_count = ceil(total_cases / BATCH_SIZE)
+print(f"Total of {total_cases} cases to process over {batch_count} batches")
+
+def process_batch(batch_number):
+    start_count = batch_number * BATCH_SIZE
+    percent_complete = int(start_count / total_cases)
+    print(f"Processing {start_count}-{BATCH_SIZE * (batch_number + 1)} ({percent_complete}%)")
+
+    # Get all relevant cases
+    closed_or_old_case_ids = (
```

Comment:
I would add `.sort("opened_on")` to ensure that each chunk is contiguous.

Reply:
That is a good idea :+1: